pkg/component/component.go

```diff
@@ -281,30 +281,78 @@ func List(client *occlient.Client, applicationName string, projectName string) (
 // GetComponentSource what source type given component uses
 // The first returned string is component source type ("git" or "local" or "binary")
 // The second returned string is a source (url to git repository or local path or path to binary)
-func GetComponentSource(client *occlient.Client, componentName string, applicationName string, projectName string) (string, string, error) {
+func GetComponentSource(client *occlient.Client, componentName string, applicationName string, projectName string) (string, string, string, error) {
 	bc, err := client.GetBuildConfig(componentName, projectName)
 	if err != nil {
-		return "", "", errors.Wrapf(err, "unable to get source path for component %s", componentName)
+		return "", "", "", errors.Wrapf(err, "unable to get source path for component %s", componentName)
 	}
 
 	sourcePath := bc.ObjectMeta.Annotations[componentSourceURLAnnotation]
 	sourceType := bc.ObjectMeta.Annotations[componentSourceTypeAnnotation]
+	componentType := bc.ObjectMeta.Labels[componentlabels.ComponentTypeLabel]
 
 	if !validateSourceType(sourceType) {
-		return "", "", fmt.Errorf("unsupported component source type %s", sourceType)
+		return "", "", "", fmt.Errorf("unsupported component source type %s", sourceType)
 	}
 
 	log.Debugf("Source for component %s is %s (%s)", componentName, sourcePath, sourceType)
-	return sourceType, sourcePath, nil
+	return sourceType, sourcePath, componentType, nil
 }
 
 // Update updates the requested component
 // Component name is the name component to be updated
 // to indicates what type of source type the component source is changing to e.g from git to local or local to binary
 // source indicates path of the source directory or binary or the git URL
-func Update(client *occlient.Client, componentName string, to string, source string) error {
+func Update(client *occlient.Client, componentName string, applicationName, to string, source string) error {
 	var err error
 	projectName := client.GetCurrentProjectName()
+	from, _, ctype, err := GetComponentSource(client, componentName, applicationName, projectName)
```

Comment:
there is a function to return Component type -  `GetComponentType`. You shouldn't modify `GetComponentSource` just to get this information.

Reply:
@kadel Ok sorry forgot that there was a function for getting the component type